test/lifecycle.test.js

```diff
@@ -82,6 +82,77 @@ test("onremove", done => {
   })
 })
 
+test("ondestroy", done => {
+  let destroyed = false
+  app({
+    state: {
+      value: true
+    },
+    view: state => actions =>
+      state.value
+        ? h(
+            "ul",
+            {
+              oncreate: () => actions.toggle()
+            },
+            [
+              h("li"),
+              h("li", {
+                onremove(element, remove) {
+                  expect(destroyed).toBe(false)
+                  remove()
+                  expect(destroyed).toBe(true)
+                  done()
+                },
+                ondestroy() {
+                  destroyed = true
+                }
+              })
+            ]
+          )
+        : h("ul", {}, [h("li")]),
+    actions: {
+      toggle: () => state => ({ value: !state.value })
+    }
+  })
+})
+
+test("nested ondestroy", done => {
```

Comment:
Would you mind adding another test with multiple nested `ondestroy`s?  The code looks fine, but it would be a nice test to have.

Could just add a child to the `h('span', { ondestroy: ... })`, perhaps a `h('strong', {ondestroy: ...} )`?